dali/server/daldap.cpp

```diff
@@ -225,9 +226,10 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
 
         if (!authenticated)
         {
-            user->credentials().setPassword(password);
-            if (!ldapsecurity->authenticateUser(*user, &superUser) || !superUser)
+            CLdapSecManager* secmgr = dynamic_cast<CLdapSecManager*>(ldapsecurity.get());
```

Comment:
Rather than a dynamic cast, it would be cleaner if isSuperUser() was added to the interface - it is logically a generally useful function.

Reply:
@ghalliday I agree that adding this to ISecManager would be better in some ways, it's bad in others. Specifically, this would require implementation in all of the HPCC Security managers (not too big of a deal), but more problematic is that external security managers like the SCAPPS team would be broken until they add this method as well.  Every time I modify this interface I get an earful from @timothyklemm .

Reply:
@RussWhitehead I'm not opposed to changes. I just like to know about them as they're happening so I can be prepared for them.

As for this change, I agree with the general usefulness. My concern is an interface change to point releases. At least with this interface, we've generally tried to limit interface changes to future major or minor releases. That this is targeting a prior minor version adds to the complexity of managing external builds.

If it has to happen, it has to happen. It just creates headaches for external builds.

Reply:
@ghalliday @timothyklemm Opened https://track.hpccsystems.com/browse/HPCC-28283 for adding to ISecManager